peer.js

```diff
@@ -410,6 +413,45 @@ TChannelPeer.prototype.connectTo = function connectTo() {
     return self.connect(true);
 };
 
+TChannelPeer.prototype.tryConnect = function tryConnect() {
+    var self = this;
+
+    var rightNow = Date.now();
+    if (rightNow < self.nextConnAttemptTime) {
+        return;
+    }
+
+    var conn = this.getOutConnection();
+    if (!conn || conn.direction !== 'out') {
+        conn = this.connectTo();
+    }
+
+    this.waitForIdentified(conn, onIdentified);
+
+    function onIdentified(err) {
+        if (!err) {
+            self.nextConnAttemptDelay = 0;
+            self.nextConnAttemptTime = 0;
+            return;
+        }
+
+        if (self.nextConnAttemptDelay === 0) {
```

Comment:
The state is stored on a `peer` instance. There is one peer per remote `hostPort`.